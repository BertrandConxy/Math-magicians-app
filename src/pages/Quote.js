import React from 'react';

const Quote = () => {
  const Quotes = [
    {
      id: 1,
      quote: 'Mathematics is not about numbers, equations or algorithms, it is about understanding',
      author: 'William Paul',
    },
    {
      id: 2,
      quote: 'Mathematics is about practice. The more you work, the more you discover',
      author: 'Berry allen',
    },
    {
      id: 3,
      quote: 'Mathematics is the mother language of science. All the disciplines use Mathematics',
      author: 'Andrew Wilson',
    },
    {
      id: 4,
      quote: 'If you could measure the capacity of thinking of the person, use Mathematics',
      author: 'Donard Phill',
    },
  ];
  const random = Math.floor(Math.random() * 4) + 1;
  return (
    <div className="quote-container">
      {Quotes.map((Quote) => {
        if (Quote.id === random) {
          return (
            <p className="quote-text">
              &quot;
              {Quote.quote}
              &quot;
              ~
              <span className="author">{Quote.author}</span>
            </p>
          );
        }
        return false;
      })}
    </div>
  );
};

export default Quote;
