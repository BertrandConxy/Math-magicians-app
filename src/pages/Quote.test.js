import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('page renders properly', () => {
  it('renders properly', () => {
    const tree = renderer.create(<Quote />).toTree();
    expect(tree).toMatchSnapshot();
  });
});
