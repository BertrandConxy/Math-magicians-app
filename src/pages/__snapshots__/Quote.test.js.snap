// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page renders properly renders properly 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="quote-text"
        >
          "
          Mathematics is not about numbers, equations or algorithms, it is about understanding
          " ~
          <span
            className="author"
          >
            William Paul
          </span>
        </p>,
        false,
        false,
        false,
      ],
      "className": "quote-container",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "\\"",
            "Mathematics is not about numbers, equations or algorithms, it is about understanding",
            "\\" ~",
            <span
              className="author"
            >
              William Paul
            </span>,
          ],
          "className": "quote-text",
        },
        "rendered": Array [
          "\\"",
          "Mathematics is not about numbers, equations or algorithms, it is about understanding",
          "\\" ~",
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": "William Paul",
              "className": "author",
            },
            "rendered": Array [
              "William Paul",
            ],
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
